src/com/dynatrace/loadrunner/logic/CFilePatcher.java

```diff
@@ -167,15 +167,24 @@ else if (aktChar=='#')
 		String start=commentedInstruction.substring(0, insertPosition);
 		String end=commentedInstruction.substring(insertPosition);
 
-		String urlEncodedPageContext=URLEncoder.encode(pageContext, "UTF-8");
+		//String urlEncodedPageContext=URLEncoder.encode(pageContext, "UTF-8");
 
 		//Insert the call to the addDynaTraceHeader function right in front of the call-->
 		//no Comments between the call to our function and the call of the web_* function
-		String parameter="\"NA="+createTimerName();
-		if (!isClickAndScriptKeyword(keyword))
-			parameter+=";PC=" + urlEncodedPageContext;
+		String TSN = createTimerName();
+		String parameter="\"";
+		if(!TSN.isEmpty()) {
+			parameter+="TSN="+createTimerName()+";";
+		}
+		if (!isClickAndScriptKeyword(keyword)) {
+			parameter+="PC=" + pageContext;
+			parameter+=";SI=LoadRunner";
```

Comment:
define
`private static final char COMMA = ';';`
which will make building string much more clear:
`
parameter.append("PC=").append(pageContext).append(COMMA);
parameter.append("SI=").append(LOAD_RUNNER);
`
above mentioned LOAD_RUNNER (not sure if it's proper name) should also be defined as a constant String